mqt/benchviewer/src/backend.py

```diff
@@ -150,14 +153,114 @@ def createDatabase(zip_file: ZipFile):
     return database
 
 
-def read_mqtbench_all_zip():
+def read_mqtbench_all_zip(
+    skip_question: bool = False,
+    target_location: str = None,
+):
     global MQTBENCH_ALL_ZIP
-    huge_zip = Path("./static/files/qasm_output/MQTBench_all.zip")
-    print("Reading in {} ({} bytes) ...".format(huge_zip.name, huge_zip.stat().st_size))
-    with huge_zip.open("rb") as zf:
+    huge_zip_path = Path(target_location + "/MQTBench_all.zip")
+
+    try:
+        mqtbench_module_version = metadata.version("mqt.bench")
+    except Exception as e:
+        print(
+            "'mqt.bench' is most likely not installed. Please run 'pip install . or pip install mqt.bench'."
+        )
+        return False
+
+    print("Searching for local benchmarks...")
+    if (
+        os.path.isfile(huge_zip_path)
+        and len(ZipFile(huge_zip_path, "r").namelist()) != 0
+    ):
+        print("... found.")
+    else:
+        print("No benchmarks found. Querying GitHub...")
+
+        version_found = False
+        response = requests.get("https://api.github.com/repos/cda-tum/mqtbench/tags")
+        available_versions = []
+        for elem in response.json():
+            available_versions.append(elem["name"])
+
+        for possible_version in available_versions:
+            if version.parse(mqtbench_module_version) >= version.parse(
+                possible_version
+            ):
+                url = (
+                    "https://api.github.com/repos/cda-tum/mqtbench/releases/tags/"
+                    + possible_version
+                )
+
+                response = requests.get(url)
+                if not response:
+                    print(
+                        "Suitable benchmarks cannot be downloaded since the GitHub API failed. "
+                        "One reasons could be that the limit of 60 API calls per hour and IP address is exceeded."
+                    )
+                    return False
+
+                if "asset" in response.json() or "assets" in response.json():
+                    for asset in response.json()["assets"]:
```

Comment:
this does only handle the case where `"assets"` is in the response and not the case where there only is `"asset"`, right?

Reply:
done